### PR TITLE
Sk validate input bams

### PR DESCRIPTION
**Pull request by skashin**

### CODE REVIEW

File: src/java/org/broadinstitute/dropseqrna/utils/ValidateAlignedSam.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        programGroup = DropSeq.class)
+
+public class ValidateAlignedSam extends CommandLineProgram {
+
+    private static final Log log = Log.getInstance(ValidateAlignedSam.class);
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "The input SAM or BAM file(s) to validate. This argument can accept wildcards, or a file with the suffix .bam_list that contains the locations of multiple BAM files", minElements = 1)
+    public List<File> INPUT_BAM;
+
+    @Argument(shortName = StandardOptionDefinitions.SORT_ORDER_SHORT_NAME)
```
Comment (alecw) [inline src/java/org/broadinstitute/dropseqrna/utils/ValidateAlignedSam.java:59]: I never remember -- is 'optional' needed if a default value is provided?

File: src/java/org/broadinstitute/dropseqrna/utils/ValidateAlignedSam.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        this.INPUT_BAM = FileListParsingUtils.expandFileList(INPUT_BAM);
+
+        File templateBamFile = null;
+        SAMSequenceDictionary templateDictionary = null;
+
+        StringBuilder errorMsg = new StringBuilder();
+        SamReaderFactory samReaderFactory = SamReaderFactory.makeDefault();
+        for (File inputBam : INPUT_BAM) {
+            try {
+                IOUtil.assertFileIsReadable(inputBam);
```
Comment (alecw) [inline src/java/org/broadinstitute/dropseqrna/utils/ValidateAlignedSam.java:76]: If this assertion failed, don't you want to skip the subsequent tests for this file?

**Review by alecw — CHANGES_REQUESTED** [review]
Hi Seva,

Just a couple of questions.  Thanks!

-Alec

**Comment by skashin** [comment]
Hi Alec,

  Good points, thanks!

Seva

**Pull request merged**